src/java.base/share/classes/java/util/zip/Adler32.java

```diff
@@ -100,7 +99,7 @@ public void update(ByteBuffer buffer) {
         if (buffer.isDirect()) {
             NIO_ACCESS.acquireSession(buffer);
             try {
-                adler = updateByteBuffer(adler, ((DirectBuffer)buffer).address(), pos, rem);
+                adler = updateByteBuffer(adler, NIO_ACCESS.getBufferAddress(buffer), pos, rem);
```

Comment:
Hello Per, from what I can see in the implementation of `NIO_ACCESS.getBufferAddress(...)` it merely returns `buffer.address` https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/nio/Buffer.java#L862. So if I understand this change correctly, what we are proposing here is that we want to avoid the additional checks (checks related to `MemorySegment` that can fail?) that are there in the `DirectBuffer.address()` method implementation? Is that what this change is addressing?

Reply:
All methods in the API that accept a ByteBuffer and access it as DirectBuffer should have been changed to use the acquireSession/getBufferAddress/releaseSession. Several were missed, I think because we didn't have enough tests to exercise the views.